main/bind/APKBUILD

```diff
@@ -174,30 +182,58 @@ dev() {
 	mv "$pkgdir"/usr/bin/isc-config.sh "$subpkgdir"/usr/bin/
 }
 
-_py3() {
-	pkgdesc="A module allowing rndc commands to be sent from Python programs"
-	depends="$_py3deps"
+plugins() {
+	pkgdesc="The ISC DNS server plugins"
+	depends="$depends_plugins"
+
+	mkdir -p "$subpkgdir"/usr/lib
+	mv "$pkgdir"/usr/lib/bind "$subpkgdir"/usr/lib/
+}
+
+py3_isc() {
+	pkgdesc="Python functions to support BIND utilities"
+	depends="$depends_py3_isc"
+	provides="py3-isc=${_isc_ver}-r${_isc_rel} py3-$pkgname=$pkgver-r$pkgrel"
```

Comment:
If I understand this correct: `py3-isc-9.14.1-r2` provides `py3-isc-2.0-r0`. In what situation will that be useful?

How is apk supposed to deal with 2 different versions of the same package without messing things up?

I can already now see problems with it. Lets say that we find a bug in `py3-isc-9.14.1-r2` so we add a  patch to the `bind` package and release `bind-9.14.1-r3`. A user upgrade one of his systems, but realizes that the patched `-r3` introduces a bad regression that will literally blow up his system. To save his other servers and prevent that someone by accident upgrade to a newer version of `py3-isc` he runs `apk add 'py3-isc<9.14.1-r3'` to make sure that he never get anything newer that `-r2`.

Now `apk` tries to resolve that, and find this `py3-isc-2.0-r1`, which is way lower than `9.14.1-r3` and installs the package providing that and *kaboom* he blows up his others systems too.

Or other potential situation: `apk add 'py3-isc>9.14'` Maybe apk will consider the provider `py3-isc-2.0` and mark that package as "does not satisfy this dependency", and thus eliminate the the `py3-isc-9.14.1-r2` from further consideration in the dependency graph.

There are just so many things that can go wrong here, and to be honest, I have no clue what apk will do in those situations, nor am I sure what I think it should do.

Reply:
> `subpackages="$pkgname-doc $pkgname-openrc py3-${pkgname}-isc:py3_isc:noarch`

This is `py3-bind-isc-9.14.1-r2` providing `py3-isc-2.0-r10` so that you can maintain the py3-isc version and ensure that its release is incremented when the source version of bind that bundles it changes.